### DISCUSSION TITLE
Hello, I am a novice from China. First of all, I am very happy to discover this library. In my opinion, it is very powerful. Unfortunately, my English is not very good, and I cannot understand the translated documents, which makes it difficult for me to get started

### DISCUSSION THREAD

**Discussion by 90757665** [pschatzmann/arduino-audio-tools#841] [Q&A] [answered]
I have encountered a problem. I have an ESP32 board that needs to be used to read the speaker sound data of another module (which only exposes two pins and can play audio when connected to a 2W speaker).
At first, I successfully obtained the PCM audio data through i2s and ADC without using any libraries and sent it to the server through websocket, but it was very noisy. I have tried demo: base adc serial to successfully view the audio data in the serial port. May I know how to convert the audio to PCM or other audio formats and ultimately obtain char data? Because I need WebsocketsClient. sendBinary() to connect to the server.
Thank you in advance, I am already exhausted.

**Comment by pschatzmann** [comment, marked as answer]
Why don't you just read the Wiki: everything you are asking is already explained!

**Reply by 90757665** [reply]
I'm really looking, maybe my skills are really poor and I haven't found the answer to my needs. Could you please help me analyze it. thank you

**Reply by 90757665** [reply]
I hope to receive your help

**Comment by joba-1** [comment]
> I cannot understand the translated documents

I have good experience with https://www.deepl.com for translating chinese data sheets to german or english.
Have you tried this?

**Reply by 90757665** [reply]
I haven't tried this yet, I've been using Google's webpage translation. Some translations of vocabulary are not very easy to understand, I will continue to try. Thank you very much

**Comment by pschatzmann** [comment]
You need to use a proper ADC: The internal ADC will always be too noisy.
How to access the data as simple byte array is described in the first chapter.
How to trannlate it to other formats is in chapter about codecs

**Reply by 90757665** [reply]
Thank you for your reply. Do you have any suitable external ADC chip suggestions for my need to collect sound from speakers (which only have two pins)? Thank you.
